mist-api/src/routes/genes/get-post-helpers.js

```diff
@@ -0,0 +1,66 @@
+'use strict'
+
+const bodyParser = require('body-parser')
+
+const searchUtil = require('lib/util')
+
+exports.geneFinder = function(app, middlewares, isPOST = false) {
+	const models = app.get('models')
+	const helper = app.get('lib').RouteHelper.for(models.Gene)
+
+	const exactMatchFieldNames = [
+		'version',
+		'locus',
+		'old_locus',
+		'stable_id',
+	]
+
+  const specificMiddlewares = []
+  if (isPOST) {
+    specificMiddlewares.push(bodyParser.urlencoded({extended: false}))
```

Comment:
Move this logic upstream inside the `post.js` route file. This file shouldn't have to know where the input search parameters originate.